### PR TITLE
NeMo Megatron GPT Unit Tests

### PR DESCRIPTION
**Pull request by ericharper**
# What does this PR do ?

Adding unit tests for GPT

_Need to add merge and vocab files to the test tarfile_

**Collection**: NLP

# Changelog 
- Test tokenizer
- Test constructor
- Test forward

# Usage

```bash
TRANSFORMERS_OFFLINE=1 pytest tests/collections/nlp/test_gpt_model.py
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -9,8 +9,15 @@ pipeline {
     timeout(time: 2, unit: 'HOURS')
     disableConcurrentBuilds()
   }
+
   stages {
 
+    stage('nvidia-smi'){
```
Comment (MaximumEntropy) [inline Jenkinsfile:15]: Not sure if you were doing this just to see what GPUs were on the CI machine?

Comment (ericharper) [reply Jenkinsfile:15]: Yup, just handy to have in the pipeline.

**Review by MaximumEntropy — APPROVED** [review]
LGTM!

**Pull request merged**